lighting/models/product_source_line.py

```diff
@@ -380,6 +380,63 @@ def _compute_wattage_display(self):
         string="Source",
     )
 
+    invisible_fields = fields.Json(
+        compute="_compute_invisible_fields",
+    )
+
+    @api.depends("is_led", "is_integrated", "is_lamp_included", "is_max_wattage")
+    def _compute_invisible_fields(self):
+        # definition of VISIBLE fields (easier that the invisible)
+        for rec in self:
+            group_map = {
+                "0000": {"wattage": False, "ccts": False, "led_specs": False},
+                "0001": {"wattage": True, "ccts": False, "led_specs": False},
+                "0010": {"wattage": True, "ccts": True, "led_specs": False},
+                "0011": {"wattage": False, "ccts": True, "led_specs": False},
+                "0100": {"wattage": True, "ccts": True, "led_specs": False},
+                "0101": {"wattage": False, "ccts": True, "led_specs": False},
+                "0110": {"wattage": True, "ccts": True, "led_specs": False},
+                "0111": {"wattage": False, "ccts": True, "led_specs": False},
+                "1000": {"wattage": False, "ccts": False, "led_specs": False},
+                "1001": {"wattage": True, "ccts": False, "led_specs": False},
+                "1010": {"wattage": True, "ccts": True, "led_specs": True},
+                "1011": {"wattage": False, "ccts": True, "led_specs": True},
+                "1100": {"wattage": True, "ccts": True, "led_specs": True},
+                "1101": {"wattage": False, "ccts": True, "led_specs": True},
+                "1110": {"wattage": True, "ccts": True, "led_specs": True},
+                "1111": {"wattage": False, "ccts": True, "led_specs": True},
+            }
+
+            field_group_map = {
+                "wattage": ["wattage", "wattage_magnitude"],
+                "ccts": [
+                    "efficiency_ids",
+                    "color_temperature_flux_ids",
+                    "is_color_temperature_flux_tunnable",
+                ],
+                "led_specs": [
+                    "cri_min",
+                    "color_consistency",
+                    "special_spectrum_id",
+                    "leds_m",
+                    "led_chip_ids",
+                ],
+            }
+
+            key_fields = [
+                "is_led",
+                "is_integrated",
+                "is_lamp_included",
+                "is_max_wattage",
+            ]
```

Comment:
Put this outside "for rec in self", maybe outside the class as a constant? What do you think?